core/include/scipp/core/transform.h

```diff
@@ -272,11 +272,15 @@ struct TransformSparse {
   }
 };
 
-template <class... Args>
-constexpr static bool
-check_all_or_none_variances(const Args &... valAndVariances) {
-  return !((valAndVariances && ...) || (!valAndVariances && ...));
-}
+template <class Op, class... Args> struct check_all_or_none_variances {
+  static const bool value =
+      std::is_base_of_v<transform_flags::expect_all_or_none_have_variance_t,
+                        Op> &&
+      is_AllOrNoneValuesAndVariances_v<Args...>;
```

Comment:
Can't you just to what `is_AllOrNoneValuesAndVariances_v` here inline? Avoids some lines of code.

Reply:
Can be done inline and would save a few lines of code I put it in `values_and_variances.h` as it seems that that it is logically related to existing type traits there.